datastore/db/bytebase/FertiScan/get_inspection/get_sub_label.sql

```diff
@@ -1,33 +1,30 @@
 CREATE OR REPLACE FUNCTION "fertiscan_0.0.14".get_sub_label_json(label_info_id uuid)
- RETURNS jsonb
- LANGUAGE plpgsql
+RETURNS jsonb
+LANGUAGE plpgsql
 AS $function$
-declare
+DECLARE
 	result_json jsonb;
-    fr text[] ;
-    en text[] ;
 BEGIN
     SELECT jsonb_object_agg(
                 sub_type.type_en, 
                 jsonb_build_object(
 				    'en', COALESCE(sub_label_data.texts_en, array[]::text[]),
 				    'fr', COALESCE(sub_label_data.texts_fr, array[]::text[])
-				)			
+				)
            )
     INTO result_json
     FROM sub_type
-    left JOIN (
+    LEFT JOIN (
         SELECT 
             sub_type_id,
-            array_agg(sub_label.text_content_en ORDER BY sub_label.id) AS texts_en,
-            array_agg(sub_label.text_content_fr ORDER BY sub_label.id) AS texts_fr
+            array_agg(sub_label.text_content_en ORDER BY sub_label.id) FILTER (WHERE sub_label.text_content_en IS NOT NULL) AS texts_en,
```

Comment:
Wont this produce problem (not returning rows) if the content_en is null but not the content_fr?

Reply:
I used the filter to avoid such arrays:
```json
{"en": ["label", null], "fr": ["label", "label"]}
```
Instead it will be:
```json
{"en": ["label"], "fr": ["label", "label"]}
```

The code in the jsonb_build_object seems to only avoid:
```json
{"en": null, "fr": ["label", "label"]}
```
Instead it will be:
```json
{"en": [], "fr": ["label", "label"]}
```

Incorrect?

Reply:
While reading the code I was under the impression that if the following exists:

| id|text_en | text_fr |
| --|------ | ------- |
| 1|text | texte |
| 2|text | Null |
| 3|Null | texte |
| 4 (should not exists in theory) | Null | Null |

the filter would only return the row with rowID=1 by blocking a row if it has a null? 

If the filter still returns all the rows with rowID= [1,2,3] this sits good with me.

I also feel it is important to return a null because if the following exists for an inspection:

| id|text_en | text_fr |
| --|------ | ------- |
| 1 | text1 | texte1 |
| 2 | text2 | Null |
| 3 | text3| texte3 |

the arrays would look like

```{"en": [text1,text2,text3], "fr": ["texte1", "text3"]}``` 

If returned for an update (with no modification) then the db will be updated as such:

| id|text_en | text_fr |
| --|------ | ------- |
| 1 | text1 | texte1 |
| 2 | text2 | texte3 |
| 3 | text3| ??? |

this is if the db doesnt throw an exception since we are looping both array until we reach the greatest index. In this case the biggest index is 3 and the fr array has 2 values. I'm not so sure about this working, ence why I'd like to see those tests cases tested with the whole pipeline 

>I see we modified new_inspection, however, I dont see a new test showcasing it can handle different type of Null combination

maybe also add in a test with the specified edge case that I mentionned with new_inspection and update_inspection to make sure the whole process behave as intented.

We might sit with the FE to make sure they can handle Null value. 

Otherwise a solution I see rolling back to `Default ''` in the schema, this might solve the edge case if there is problems

Reply:
This:
```json
{"en": ["text1","text2","text3"], "fr": ["texte1", "text3"]}
```
was per your requirements.

`fr_value := fr_values->>i;` returns null for an overflow, not an exception.

I'm fine with returning a placeholder, because it preserves the correct positions between en and fr. I suggest the placeholder be an empty string instead of null however. Indeed, in the frontend we have this:
```typescript
export const populateFromJSON = (form: Data, data: Inspection) => {
  form.sections.forEach((section) => {
    section.inputs.forEach((input) => {
      const value = access_from_str(data, input.id);
      if (typeof value == "string") {
        input.value = [value];
      } else if (Array.isArray(value)) {
        if (value.length == 0) {
          input.isAlreadyTable = true;
          input.value = [""];
        } else if (typeof value[0] == "string") {  //<---------------- here
          input.value = value;
          input.isAlreadyTable = true;
        } else if (typeof value[0] == "object") {   //<---------------- and here
          input.value = value;
          input.isInputObjectList = true;
        }
      } else if (typeof value == "object") {
        // eslint-disable-next-line
        // @ts-ignore
        input.value = [value];
      }
    });
  });
  // for state update the function must return a new object
  return form.copy();
};
```

If ever the first checked value is null, the rest will be ignored.

Reply:
>fr_value := fr_values->>i; returns null for an overflow, not an exception.

So the db will be 

id | text_en | text_fr
-- | -- | --
1 | text1 | texte1
2 | text2 | texte3
3 | text3 | Null

Instead of being

id | text_en | text_fr
-- | -- | --
1 | text1 | texte1
2 | text2 | Null
3 | text3 | texte3

>was per your requirements.

...?
I think you are referring to
>An empty string is still a value that will be registered and does not represent the Edge case we are attempting to solve.

What I mean is, we do not want to save an empty string in the database when there are no record of a value being there.
>I'm fine with returning a placeholder, because it preserves the correct positions between en and fr. I suggest the placeholder be an empty string instead of null

I'm also ok with returning an empty string as a place holder. This could also be used by the frontend to detect empty fields that should be flagged to the inspector. However, We must check, once we receive it back that the update is performed correctly since the update is actually a delete + insert .


- Case 1: 
 -> The place holder returned as some text (length>0)
l---> New.text_content = some text
- Case 2: 
-> The returned value of a sub_label_string is an empty string (length=0)
l---> New.text_content = Null

Reply:
Not sure I understand the last part. I've already updated the update_sub_labels function and tested it.